engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -289,13 +313,23 @@ class EngineJobExecutionActor(replyTo: ActorRef,
       stay
   }
 
+  private def publishHashesToMetadata(maybeHashes: Option[Try[CallCacheHashes]]) = maybeHashes match {
+    case Some(Success(hashes)) =>
+      val hashMap = hashes.hashes.collect({
+        case HashResult(HashKey(useInCallCaching, keyComponents), HashValue(value)) if useInCallCaching => (callCachingHashes + keyComponents.mkString(MetadataKey.KeySeparator.toString)) -> value
+      }).toMap
+      writeToMetadata(hashMap)
+    case _ => log.error("Cannot find hashes although they were successfully written to the database.")
```

Comment:
Good catch :) If we have a failure would you expect to see it in the metadata still in place of the hashes ?

```
{
...
"hashes": "Failed to get hashes because..."
...
}
```

It's harder to parse for a client though because the structure is different depending on success/failure